src/persistence/secondary/mongo-secondary-entity-manager.ts

```diff
@@ -1,77 +1,71 @@
 import { MongoBuildArgs } from '../../model/mongo-build-args';
 import { MongoModel } from '../../model/mongo-model';
 import { SecondaryEntityManager } from './secondary-entity-manager';
 
 export class MongoSecondaryEntityManager<TEntity extends Entity> implements SecondaryEntityManager<TEntity> {
-  public get model(): MongoModel<TEntity> {
-    return this._model;
-  }
-
+  protected _db: Promise<Db>;
   protected _model: MongoModel<TEntity>;
-  private _collection: Promise<Collection>;
   private _client: Promise<MongoClient>;
 
-  constructor(model: MongoModel<TEntity>, args: MongoBuildArgs) {
+  public constructor(model: MongoModel<TEntity>, args: MongoBuildArgs) {
     this._model = model;
     this._client = MongoClient.connect(args.url);
-    this._collection = this._client.then((client) => client.db(args.dbName).collection(this._model.collectionName));
+    this._db = this._client.then((client) => client.db(args.dbName));
+  }
+
+  public get model(): MongoModel<TEntity> {
+    return this._model;
   }
 
   public async delete(id: string | number): Promise<any> {
-    return (await this._collection).findOneAndDelete({ id: id });
+    return (await this._getCollection()).findOneAndDelete({ id: id });
   }
 
   public async insert(entity: TEntity): Promise<any> {
-    return (await this._collection).insertOne(entity);
+    return (await this._getCollection()).insertOne(entity);
   }
 
   public async mDelete(ids: string[] | number[]): Promise<any> {
-    return (await this._collection).deleteMany({ id: { $in: ids } });
+    return (await this._getCollection()).deleteMany({ id: { $in: ids } });
   }
 
   public async mInsert(entities: TEntity[]): Promise<any> {
-    return (await this._collection).insertMany(entities);
+    return (await this._getCollection()).insertMany(entities);
   }
 
-  public async mUpdate(entities: TEntity[]): Promise<any> {
-    const session = (await this._client).startSession();
-    try {
-      await session.withTransaction(async () => {
-        const results: Array<Promise<any>> = [];
-        for (const entity of entities) {
-          results.push((await this._collection).findOneAndUpdate({ id: entity.id }, { $set: entity }));
-        }
-        return Promise.all(results);
-      });
-    } finally {
-      await session.endSession();
-    }
+  public mUpdate(entities: TEntity[]): Promise<any> {
+    return this._getCollection().then((collection) => {
+      return collection.bulkWrite(
+        entities.map((entity) => {
+          return { updateOne: { filter: { id: entity.id }, update: { $set: entity } } };
+        }),
+        { ordered: false },
+      );
+    });
   }
 
   public async update(entity: TEntity): Promise<any> {
-    return (await this._collection).findOneAndUpdate(
-      { id: entity.id },
-      { $set: entity },
-      {
-        upsert: true,
-      },
-    );
+    return (await this._getCollection()).updateOne({ id: entity.id }, { $set: entity });
```

Comment:
We just want to notify that the entity was updated, so there is no need to use findOneAndUpdate

Reply:
Oh, I though we need the updated document. then better updateOne